app/Http/Controllers/admin/TopicController.php

```diff
@@ -0,0 +1,187 @@
+     *
+     * @param  \Illuminate\Http\Request $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(TopicFormRequest $request)
+    {
+        if (Auth::check()) {
+            $user_id = Auth::user()->id;
+            $request->merge(['user_id' => $user_id]);
+            $image = $request->file('image_topic');
+            $img = $this->saveImage($image);
+            $request->merge(['images' => $img]);
+
+            //handle select time
+            if ($request->input('set_time') == 0) {
+                $select_time = null;
+            } elseif ($request->input('set_time') == 1) {
+                $select_time = $request->input('select_time');
+
+            }
+            $request->merge(['select_time' => $select_time]);
+//            dd(Carbon::now()->hour . ':'. Carbon::now()->minute);
+            $this->topic->store($request->all());
+
+            return back()->with('success', __('message.success'));
+        }
+
+    }
+
+    /**
+     * Display the specified resource.
+     *
+     * @param  int $id
+     * @return \Illuminate\Http\Response
+     */
+    public function show($id)
+    {
+        $topic = $this->topic->show($id);
+
+        return view('admin.topic.show', compact('topic'));
+    }
+
+    /**
+     * Show the form for editing the specified resource.
+     *
+     * @param  int $id
+     * @return \Illuminate\Http\Response
+     */
+    public function edit($id)
+    {
+        $topic = $this->topic->find($id);
+
+        return view('admin.topic.edit', compact('topic'));
+    }
+
+    public function updateAll(TopicFormRequest $request, $id)
+    {
+        $topic = $this->topic->find($id);
+        $image = $request->file('image_topic');
+        $img = $this->saveImage($image);
+        $request->merge(['images' => $img]);
+        $topic->update($request->all());
+
+        return back()->with('success', trans('message.success'));
+    }
+
+    /**
+     * Update the specified resource in storage.
+     *
+     * @param  \Illuminate\Http\Request $request
+     * @param  int $id
+     * @return \Illuminate\Http\Response
+     */
+    public function update(Request $request, $id)
+    {
+        $test = Topic::find($id);
+        $column_name = Input::get('name');
+        $column_value = Input::get('value');
+        if (Input::has('name')) {
+            $test = Topic::select()
+                ->where('id', '=', $id)
+                ->update([$column_name => $column_value]);
+
+            return response()->json(['code' => 200], 200);
+        }
+
+        return response()->json(['error' => 400, 'message' => 'Not enought params'], 400);
+    }
+
+    public function bulk_update(Request $request)
+    {
+        if (Input::has('ids_to_edit') && Input::has('bulk_name') && Input::has('bulk_value')) {
+            $ids = Input::get('ids_to_edit');
+            $bulk_name = Input::get('bulk_name');
+            $bulk_value = Input::get('bulk_value');
```

Comment:
**Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Variables.ValidVariableName.NotCamelCaps</br>**Severity**: error</br>**File**: app/Http/Controllers/admin/TopicController.php</br>**Line**: 152</br>Variable "bulk_value" is not in valid camel caps format